### ISSUE TITLE
Finetuning Training Time

### ISSUE THREAD

**Issue by madhavajay** [chuanqi305/MobileNet-SSD#96]
I have a few questions about performance and training time etc.

I have the Caffe SSD compiled with CUDA 9 and cuDNN 7 plus OpenBlas and pretty much every optimisation I could see except these:
https://github.com/chuanqi305/MobileNetv2-SSDLite/tree/master/src

1) Are those optimisations relevant for MobileNet-SSD v1? and are they worth re-compiling it with?
2) I am using a px2.large with a K80
[0] Tesla K80        | 53'C,  75 % |  5889 / 11441 MB | madhavajay(5876M)
madhavajay@ip-172-31-36-146:~/MobileNet-SSD/snapshot$ gpustat
ip-172-31-36-146  Fri Jun  1 01:15:39 2018
[0] Tesla K80        | 53'C,  82 % |  5889 / 11441 MB | madhavajay(5876M)
madhavajay@ip-172-31-36-146:~/MobileNet-SSD/snapshot$ gpustat
ip-172-31-36-146  Fri Jun  1 01:15:42 2018
[0] Tesla K80        | 53'C,  68 % |  5889 / 11441 MB | madhavajay(5876M)

It seems like I should be able to up the batch size.
I tried going from 24 to 32 but the time per 10 iterations went from 45 seconds to 1 minute.
Is that just because the intervals had more data and therefore its better? I thought an iteration should be the entire data set in which case why is it slower with a higher batch size even though the GPU isnt full?

Can anyone recommend optimal settings?
3) Is it really that slow, im just training on 9 categories from COCO and its taking 45 seconds to 1 minute for 10 iterations? The recommended number is 120,000 that would take 200 hours on a single GPU. How will it scale if I throw 4 of these GPUs at it?

4) I am using the mobilenet_iter_73000.caffemodel for fine tuning so should this be a lot faster to converge? Is there a good way to know, e.g. by looking at the loss?

5) I have 9 classes (including background) but have 13 in my file as i have put in Blank ones to keep the label int numbers sequential, is there a way around this? The data I have from coco is in the original label int order and I dont want to change the mappings there. But the extra unused classes add to the model size and performance.

Thanks.

**Comment by BigMuscle85** [comment]
MobileNet-SSD is here really slow during training, even on GPU. With basic ResNet-10 SSD network, it takes about 40 seconds per 100 iterations and 140,000 iterations just finishes over night. MobileNet-SSD takes about 6 minutes per 100 iterations and 120,000 does not finish over weekend. And final accuracy is almost the same.
